src/test/scala/com/jcdecaux/setl/storage/repository/SparkRepositorySuite.scala

```diff
@@ -129,6 +129,59 @@ class SparkRepositorySuite extends AnyFunSuite with Matchers {
     assert(!connector2.read().columns.contains("partition"))
   }
 
+  test("SparkRepository should handle update (upsert) when ACIDConnector is used") {
+    val spark: SparkSession = new SparkSessionBuilder().setEnv("local").build().get()
+    assume(SparkTestUtils.checkSparkVersion("2.4.2"))
+
+    import spark.implicits._
+
+    val ds: Dataset[TestDeltaUpdate] = Seq(TestDeltaUpdate(1, "c1", 2.3D), TestDeltaUpdate(2, "c2", 4.9D)).toDS()
+    val dss: Dataset[TestDeltaUpdate] = Seq(TestDeltaUpdate(1, "c1bis", 4.4D), TestDeltaUpdate(1, "c1", 5.1D)).toDS()
+    val path: String = "src/test/resources/test_delta_with_anno"
+    val connector = new DeltaConnector(Map[String, String](
+      "path" -> path,
+      "saveMode" -> "Append"
+    ))
+    val condition = Condition("col1", "=", 2)
+
+    val repo = new SparkRepository[TestDeltaUpdate].setConnector(connector)
+
+    connector.drop()
+
+    repo
+      .partitionBy("_sort_key")
+      .save(ds)
+
+    // to verify if the filter is pushed down
+    val filteredBySortKey = repo.findBy(Condition("_sort_key", "=", "c2"))
+    filteredBySortKey.explain()
```

Comment:
maybe we can check if the query plan string contains the name of the pushed down filters. I did this in `ParquetConnectorSuite`. You can take a look at the test "Parquet connector should push down filter and select"'